src/main/java/sg/ncl/MainController.java

```diff
@@ -1476,6 +1483,34 @@ public String teamProfile(@PathVariable String teamId, Model model, HttpSession
         model.addAttribute("teamExperimentList", experimentList);
         model.addAttribute("teamRealizationMap", realizationMap);
 
+        //Starting to get quota
+        try {
+            response = restTemplate.exchange(properties.getQuotaByTeamId(teamId), HttpMethod.GET, request, String.class);
+        } catch (RestClientException e) {
+            log.warn("Error connecting to sio team service for display team quota: {}", e);
+            redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+            return "redirect:/team_profile/{teamId}";
+        }
+
+        responseBody = response.getBody().toString();
+
+        // handling exceptions from SIO
+        if (RestUtil.isError(response.getStatusCode())) {
+            MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+            ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+            switch (exceptionState) {
+                case TEAM_NOT_FOUND_EXCEPTION:
+                    log.warn("Get team quota: Team {} not found", teamId);
+                default:
+                    log.warn("Get team quota : sio or deterlab adapter connection error");
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+                    break;
+            }
+        }
+
+        TeamQuota teamQuota = extractTeamQuotaInfo(responseBody);
+        model.addAttribute("teamQuota", teamQuota);
+        session.setAttribute("originalBudget", teamQuota.getBudget()); // this is to check if budget changed later
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "originalBudget" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)